### PR TITLE
Allow animal-drawn vehicles to follow a walking player

### PR DESCRIPTION
**Pull request by ghost**
#### Summary
```SUMMARY: Features "Allow animal-drawn vehicles to follow a walking player"```

#### Purpose of change
Allow animal-drawn vehicles to follow a walking player

#### Describe the solution
From Chat menu `C` , two new shouts, if you have animal-drawn vehicles nearby, they are whistle to tell them to follow, or whistle to tell them to stop.

The horse-drawn vehicle will then follow you at a safe distance ( although, like with autodriving - huge vehicles are gonna cause problems )

It will only use the animal "engine" and turn off the others.
If you board the vehicle in question, it will stop.
if it encounters obstacles, it will stop.
Therefore, travelling on roads or empty fields will be preferable.

#### Describe alternatives you've considered
The next phase of this, is a part like the Robot Controls or drive-by-wire, that allows normal vehicles to follow player.
Then also it could be expanded, for NPC driving to follow in a convoy, or un-manned vehicle drones on patrol paths etc, NPC caravans and so on.
#### Additional context
A gif of it in action https://gfycat.com/wildearnestkawala

bonus gif of in-progress hilarity when the wheelchair wouldnt stop following me untl I was dead.

https://gfycat.com/brightenormousdutchshepherddog

### CODE REVIEW

File: src/npctalk.cpp [hunk 1]
```diff
@@ -338,6 +365,17 @@ void game::chat()
         g->u.add_msg_if_player( m_warning, _( "You can't speak without your face!" ) );
         return;
     }
+    std::vector<vehicle *> animal_vehicles;
+    std::vector<vehicle *> following_vehicles;
+    for( auto &veh : g->m.get_vehicles() ) {
+        auto &v = veh.v;
+        if( v->has_engine_type( fuel_type_animal, false ) ) {
```
Comment (BevapDin) [inline src/npctalk.cpp]: There is no owner check here like in `assign_veh_to_follow`, is that intended? (The player can issue the command even if there is no suitable vehicle around, only non-owned vehicles with animals may be around.)

File: src/vehicle.h [hunk 2]
```diff
@@ -1695,6 +1703,7 @@ class vehicle
 
     public:
         bool is_autodriving = false;
+        bool is_following = false;
```
Comment (BevapDin) [inline src/vehicle.h]: This is not stored in the save. See savegame_json.cpp for storing and loading it.

Comment (davidpwbrown) [reply src/vehicle.h]: oops yeah.

**Comment by adamkad1** [comment]
now i wont have to pull my shopping cart myself!

**Pull request merged**